File: frontend/src/types/system.ts
export interface SystemHealthResponse {
    overall_status: string;
    cpu_usage: number;
    memory_usage: number;
    disk_usage: number;
    uptime_seconds: number;
    process_memory_mb: number;
    directories_status: Record<string, any>;
    services_status: Record<string, string>;
    health_issues: string[];
    last_check: string;
  }
  
  export interface SystemStatsResponse {
    system_info: Record<string, any>;
    network_stats: Record<string, number>;
    data_stats: Record<string, any>;
    recent_activity: Record<string, any>;
    generated_at: string;
  }
  
  export interface ConfigurationRequest {
    configuration: Record<string, any>;
  }
  
  export interface UserRequest {
    username: string;
    email: string;
    role: string;
  }
  
  export interface UserResponse {
    username: string;
    email: string;
    role: string;
    active: boolean;
    created_at: string;
    last_login?: string;
  }
  
  export interface MaintenanceRequest {
    task_type: string;
    parameters?: Record<string, any>;
  }

File: web/models/system.py
from pydantic import BaseModel
from typing import Dict, List, Any, Optional
from datetime import datetime

class SystemHealthResponse(BaseModel):
    overall_status: str
    cpu_usage: float
    memory_usage: float
    disk_usage: float
    uptime_seconds: int
    process_memory_mb: float
    directories_status: Dict[str, Any]
    services_status: Dict[str, str]
    health_issues: List[str]
    last_check: datetime

class SystemStatsResponse(BaseModel):
    system_info: Dict[str, Any]
    network_stats: Dict[str, int]
    data_stats: Dict[str, Any]
    recent_activity: Dict[str, Any]
    generated_at: datetime

class ConfigurationRequest(BaseModel):
    configuration: Dict[str, Any]

class ConfigurationResponse(BaseModel):
    configuration: Dict[str, Any]
    last_updated: datetime
    updated_by: str

# class UserRequest(BaseModel):
#     username: str
#     email: str
#     role: str

# class UserResponse(BaseModel):
#     username: str
#     email: str
#     role: str
#     active: bool
#     created_at: datetime
#     last_login: Optional[datetime] = None

# class MaintenanceRequest(BaseModel):
#     task_type: str
#     parameters: Optional[Dict[str, Any]] = None